### PR TITLE
docs: add dedicated TypeOrValueSpecifier docs page

### LINKED ISSUE #9587: Docs: Add a dedicated docs page for type-utils > TypeOrValueSpecifier
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

The [`TypeOrValueSpecifier`](https://github.com/typescript-eslint/typescript-eslint/blob/90baceea968c6e10917feec19bc063c8bb9919b6/packages/type-utils/src/TypeOrValueSpecifier.ts) utility is used by a couple a few rules now:

* `no-floating-promises`: #9186 and #9234
* `prefer-readonly-parameter-types`: #4436

...and is pending for:

* `no-unnecessary-type-parameters`: #9793
* `restrict-template-expressions`: #8556 -> #8722

We've been saying constantly that we want shared documentation for this thing. @abrahamguo's https://github.com/typescript-eslint/typescript-eslint/pull/8556/files#diff-3af7b11cfbfbd4c66519dff13130df296bdf564fbd1056765f1c79dbed9c2485 is a good start. We'll want to add a true docs page for it with the team's recommendations, including:

* All the things already mentioned in those docs
* To avoid using it as an `allow` option whenever possible (+ why)
* That there is a single `string` allowed format, but it's deprecated, and will likely eventually be removed (+ why)

### Affected URL(s)

https://typescript-eslint.io/packages/type-utils/type-or-value-specifier, maybe?

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
## PR Checklist

- [x] Addresses an existing open issue: fixes #9587
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

Adds a dedicated docs page, cross-linked with the generated TypeDocs and each rule options page.

I mentioned everything in the issue except the recommendation to not use this. I think the individual rule options do a more convincing job of advocating for that.

💖

### CODE REVIEW

File: packages/website/src/css/custom.css [hunk 1]
```diff
@@ -204,6 +204,10 @@ h6 {
   text-decoration: underline;
 }
 
+.markdown a:has(code) {
+  text-underline-offset: 0.25em;
+}
```
Comment (JoshuaKGoldberg) [inline packages/website/src/css/custom.css:209]: This was really bugging me...

<table>
<thead>
<tr>
<th>Before</th>
<th>After</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Before screenshot of the links in Rule Options Using This Format, with almost no vertical padding above text underlines
</td>
<td>
After screenshot of the links in Rule Options Using This Format, with several pixels of vertical padding above text underlines
</td>
</tr>
</tbody>
</table>

File: docs/packages/type-utils/TypeOrValueSpecifier.mdx
Comment (JoshuaKGoldberg) [inline docs/packages/type-utils/TypeOrValueSpecifier.mdx:1]: fyi @abrahamguo (e.g. #8556) & @RebeccaStevens & @marekdedic 🙂

Comment (JoshuaKGoldberg) [inline docs/packages/type-utils/TypeOrValueSpecifier.mdx:1]: fyi @abrahamguo & @marekdedic & @RebeccaStevens 🙂

File: docs/packages/type-utils/TypeOrValueSpecifier.mdx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+interface PackageSpecifier {
+  from: 'package';
+  name: string[] | string;
+  package: string;
+}
+```
+
+Describes specific types or values imported from packages.
+
+`path` must be used to specify the package name.
```
Comment (marekdedic) [inline docs/packages/type-utils/TypeOrValueSpecifier.mdx:103]: Should probably be `package`

**Review by marekdedic — APPROVED** [review]
Thanks, LGTM!

File: packages/website/src/css/custom.css (hunk 1 above)
Comment (Josh-Cena) [reply packages/website/src/css/custom.css:209]: Hmm. I really don't have a preference.

**Review by Josh-Cena — DISMISSED** [review]
Looks good

**Pull request merged**